### PR TITLE
MINOR: reuse decompression buffers in log cleaner

### PR DESCRIPTION
**Pull request by xvrl**
follow-up to KAFKA-5150, reuse decompression buffers in the log cleaner thread.

@ijuma @hachikuji

### CODE REVIEW

**Review by hachikuji — APPROVED** [review]
LGTM. I'll let @ijuma decide where this should go.

File: clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java:134]: Can you please add a javadoc for the `decompressionBufferSupplier` parameter?

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM, just one minor comment.

**Comment by ijuma** [comment]
I'll add the comment before merging to trunk and 0.11.0.

File: clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java
Comment (xvrl) [reply clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java:134]: done

**Comment by ijuma** [comment]
@xvrl updated the PR, so I just merged as is.

**Pull request closed without merging**
